momepy/utils.py

```diff
@@ -47,7 +48,14 @@ def _angle(a, b, c):
     return abs((a2 - a1 + 180) % 360 - 180)
 
 
-def _generate_primal(graph, gdf_network, fields, multigraph, oneway_column=None):
+def _make_edgetuples(nodelist):
+    """Zip list of nodes into list of edgetuples. Helper function for _generate_primal"""
+    return [z for z in zip(nodelist, nodelist[1:])]
```

Comment:
Can we make this more robust with?:

```suggestion
    return [z for z in zip(nodelist[:-1], nodelist[1:], strict=True)]
```